src/legacy/core_plugins/console/public/np_ready/application/containers/editor/legacy/console_editor/editor.tsx

```diff
@@ -81,10 +86,54 @@ function EditorUI() {
   useEffect(() => {
     editorInstanceRef.current = senseEditor.create(editorRef.current!);
 
-    const { content: text } = history.getSavedEditorState() || {
-      content: DEFAULT_INPUT_VALUE,
+    const getQueryParams = () => {
+      const [, queryString] = window.location.hash.split('?');
+      return qs.parse(queryString || '');
     };
-    editorInstanceRef.current.update(text);
+
+    const onHashChange = debounce(() => {
+      const qParams = getQueryParams();
+      if (!qParams || !qParams.load_from) {
+        return;
+      }
+      const sourceLocation: string = qParams.load_from;
+      if (/^https?:\/\//.test(sourceLocation)) {
+        const loadFrom: Record<string, any> = {
+          url: sourceLocation,
+          // Having dataType here is required as it doesn't allow jQuery to `eval` content
+          // coming from the external source thereby preventing XSS attack.
+          dataType: 'text',
+          kbnXsrfToken: false,
+        };
+
+        if (/https?:\/\/api\.github\.com/.test(sourceLocation)) {
+          loadFrom.headers = { Accept: 'application/vnd.github.v3.raw' };
+        }
+
+        $.ajax(loadFrom).done(async data => {
+          await editorInstanceRef.current!.update(data);
+          editorInstanceRef.current!.moveToNextRequestEdge(false);
+          editorInstanceRef.current!.highlightCurrentRequestsAndUpdateActionBar();
+          editorInstanceRef
+            .current!.getCoreEditor()
+            .getContainer()
+            .focus();
+        });
+      }
+    }, 200);
+
+    window.addEventListener('hashchange', onHashChange);
+
+    const firstParamsCheck = getQueryParams();
+    if (firstParamsCheck && firstParamsCheck.load_from) {
```

Comment:
This check looks like a duplicate of the check on line 96. Maybe we only need one or the other?

Reply:
So the two checks are for different points in time:
1. Startup - when we want to figure out should we load from remote or should we load from localStorage
2. Later, when hash changes occur, should we load from remote - we don't want to load from localStorage if not from remote in this case.

I guess the alternative could be to set a `Ref` and check if we have done the "initial load". But this way can be restructured to only do the check once